README.md

```diff
@@ -8,6 +8,265 @@
+input of type `checkbox`. Other than that, everything remains the same.
+
+Wouldn't it be nice if we could de-duplicate this logic and extract out the
+redundant `handleClick` handler?
+
+### Modularizing actions
+
+If we look at the code of the toggle button and chocolate disclaimer, we notice
+that the click handlers are completely **identical**.
+
+Code redundancy is never a good thing, therefore we should try to extract out
+our `handleClick` function.
+
+One way would be to have a super-class `Toggleable` that both
+`ChocolateDisclaimer` and `ToggleButton` inherit from:
+
+```js
+class Toggleable extends React.Component {
+  toggleState () {
+    this.setState({
+      enabled: !this.state.enabled
+    });
+  }
+}
+```
+
+`ToggleButton` could then inherit from this superclass:
+
+```js
+class ToggleButton extends Toggleable {
+  constructor (props) {
+    super(props);
+    this.state = {
+      // initially our `ToggleButton` is 
+      enabled: false
+    };
+  }
+  render () {
+    return (
+      <button onClick={this.toggleState}>
+        {this.state.enabled ? 'Enabled' : 'Disabled'}
+      </button>
+    );
+  }
+}
+```
+
+This is already a bit better, but becomes quickly unmanageable. A component
+might for instance be `Toggleable` and `Submittable`… which becomes a bit messy
+very quickly. JavaScript doesn't support multiple-inheritance (inheriting from
+more than one class at once) and building our own mixin system is a bit
+unnecessary.
+
+Instead, we can extract out our handler into a separate "action".
+
+**Advanced:** Redux also uses actions. We're trying to slowly move towards such
+a Flux-like architecture. Hold on! We'll get there!
+
+An action is just a plain function that accepts a context (`ctx`) and arbitrary
+additional arguments (for instance the event itself in case we want to
+`.preventDefault()`):
+
+```js
+function toggleState (ctx, ev) {
+  ctx.setState({
+    enabled: !ctx.state.enabled
+  });
+}
+```
+
+`ctx` is the `this` context of the component that the action is "bound" to.
+
+The above action could be used in the `ToggleButton` as well as the
+`ChocolateDisclaimer`.
+
+In both cases, the context would be `this`. The action would be bound inside the
+component's constructor function:
+
+```js
+class ToggleButton extends React.Component {
+  constructor (props) {
+    super(props);
+    this.state = { enabled: false };
+    this.toggleState = () => toggleState(this);
+  }
+  render () {
+    // ...
+  }
+}
+```
+
+Instead of using an arrow function, we could also "curry" the `toggleState`
+function. Currying a functions means converting a function that accepts multiple
+arguments into a function that accepts one argument at a time:
+
+```js
+function toggleState (ctx, ev) {
+  ctx.setState({
+    enabled: !ctx.state.enabled
+  });
+}
+```
+
+could also be written as
+
+```js
+const toggleState => ctx => ev =>
+  ctx.setState({
+    enabled: !ctx.state.enabled
+  });
```

Comment:
Can we use one implicit return here for the outer function to improve readability? I feel like chaining two shorthand arrow functions kind of impedes on the ability to quickly scan the code. Might just be me though. 😄